### ISSUE TITLE
Failed to start a connection

### ISSUE THREAD

**Issue by miabrahams** [TelepathyQt/telepathy-morse#10]
I have not been able to get a connection up and running with Empathy or KDE Telepathy.  It seems like the protocol and account are being properly registered with D-Bus, but no connection is being made. In .xsession-errors, I see: 

```
../../../../lib/SignOn/identityimpl.cpp 381 getInfoReply QMap(("ACL", QVariant(QStringList, ("*")) ) ( "AuthMethods" ,  QVariant(QDBusArgument, ) ) ( "Caption" ,  QVariant(QString, "ktp-morse-telegram") ) ( "Id" ,  QVariant(uint, 11) ) ( "Owner" ,  QVariant(QStringList, ()) ) ( "Realms" ,  QVariant(QStringList, ()) ) ( "RefCount" ,  QVariant(int, 0) ) ( "StoreSecret" ,  QVariant(bool, true) ) ( "Type" ,  QVariant(int, 1) ) ( "UserName" ,  QVariant(QString, "*My-phone-number*") ) ( "UserNameSecret" ,  QVariant(bool, false) ) ( "Validated" ,  QVariant(bool, false) ) ) 
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "Ready" SignOn::IdentityImpl(0x9317a00)
Info:
    Id: 11
    caption: "ktp-morse-telegram"
    owner: ""
    userName: "*my-phone-number*"
```

and `org.freedesktop.Telepathy.AccountManager` lists a Telegram account in its paths.  

However when I attempt to connect, I face a "There was a problem while trying to connect" error. The error logs repeat this six line message a few times, with the string `1d0b180` changing: 

```
tp-qt 0.9.6.1 WARN: Nested PendingReady for true failed with "org.freedesktop.DBus.Error.NameHasNoOwner" : "Name owner lost (service crashed?)" 
tp-qt 0.9.6.1 WARN: Building connection "/org/freedesktop/Telepathy/Connection/morse/telegram/connection_1d0b180" failed with "org.freedesktop.DBus.Error.NameHasNoOwner" - "Name owner lost (service crashed?)" 
tp-qt 0.9.6.1 WARN: StatefulDBusProxy::uniqueNameFrom(): Failed to get unique name of "org.freedesktop.Telepathy.Connection.morse.telegram.connection_1d0b180" 
tp-qt 0.9.6.1 WARN:   error: "org.freedesktop.DBus.Error.NameHasNoOwner" message: "Could not get owner of name 'org.freedesktop.Telepathy.Connection.morse.telegram.connection_1d0b180': no such name" 
tp-qt 0.9.6.1 WARN: Nested PendingReady for true failed with "org.freedesktop.DBus.Error.NameHasNoOwner" : "Could not get owner of name 'org.freedesktop.Telepathy.Connection.morse.telegram.connection_1d0b180': no such name" 
tp-qt 0.9.6.1 WARN: Building connection "/org/freedesktop/Telepathy/Connection/morse/telegram/connection_1d0b180" failed with "org.freedesktop.DBus.Error.NameHasNoOwner" - "Could not get owner of name 'org.freedesktop.Telepathy.Connection.morse.telegram.connection_1d0b180': no such name" 
```

Afterward I see a few of these with some assorted disconnect messages:
`tp-qt 0.9.6.1 WARN: Connection::capabilities() used before connection FeatureCore is ready`

Two things about my setup.  First, after the first failures I tried deleting the files and installing to /usr - when that didn't work I went back to to /usr/local.  However there were enough kbuildsycoca's and reboots in between I don't think that's an issue. Second thing is that I'm on Kubuntu 15.10 Beta.

**Comment by Kaffeine** [comment]
Hello.

Looks like DBus-activation doesn't work. Can you try to start morse manually before going online? ( `/usr/libexec/telepathy-morse` or `/usr/local/libexec/telepathy-morse` ) If there would be a morse error, rather than setup and activation problem, then I need a log from telepathy-morse terminal output.

**Comment by miabrahams** [comment]
Hi, thank you for the support. I was able to run Morse from the command line and get a little further this time.  The debug output from the command line appeared successful without any warnings.

```
MorseProtocol::MorseProtocol(const QDBusConnection&, const QString&)
tp-qt 0.9.6.1 DEBUG: Interface "org.freedesktop.Telepathy.Protocol.Interface.Addressing" plugged 
tp-qt 0.9.6.1 DEBUG: Interface "org.freedesktop.Telepathy.Protocol.Interface.Avatars" plugged 
tp-qt 0.9.6.1 DEBUG: Interface "org.freedesktop.Telepathy.Protocol.Interface.Presence" plugged 
tp-qt 0.9.6.1 DEBUG: Protocol "telegram" added to CM 
tp-qt 0.9.6.1 DEBUG: Registering protocol "telegram" at path "/org/freedesktop/Telepathy/ConnectionManager/morse/telegram" for CM "/org/freedesktop/Telepathy/ConnectionManager/morse" at bus name "org.freedesktop.Telepathy.ConnectionManager.morse" 
tp-qt 0.9.6.1 DEBUG: Registered object "/org/freedesktop/Telepathy/ConnectionManager/morse/telegram" at bus name "org.freedesktop.Telepathy.ConnectionManager.morse" 
tp-qt 0.9.6.1 DEBUG: Registering CM "/org/freedesktop/Telepathy/ConnectionManager/morse" at bus name "org.freedesktop.Telepathy.ConnectionManager.morse" 
tp-qt 0.9.6.1 DEBUG: Registered object "/org/freedesktop/Telepathy/ConnectionManager/morse" at bus name "org.freedesktop.Telepathy.ConnectionManager.morse" 
```

Similarly the debug info in .xsession-errors at launch seemed alright.

```
Starting new account dialog for "ktp-morse-telegram"
"ktp-morse-telegram"
Requested UI for manager "morse" and protocol "telegram"
tp-qt 0.9.6.1 WARN: Error parsing config file for connection manager "haze" - introspecting 
Creating service for "ktp-morse-telegram"
QVariant::QString "account"
QModelIndex(0,0,0x0,ParameterEditModel(0x111d2360) )  "account"
log_koalarmclient: Check: "Thu Oct 1 14:23:34 2015"  - "Thu Oct 1 14:24:33 2015"
kdeconnect.core: Broadcasting identity packet
```

However when I tried to launch KDE Telepathy I got another red info bubble, and the error log looked like this: 

```
../../../../lib/SignOn/connection-manager.cpp 106 setupSocketConnection p2p error: QDBusError("org.freedesktop.DBus.Error.FileNotFound", "Failed to connect to socket /run/user/1000/signond/socket: No such file or directory") 1
../../../../lib/SignOn/connection-manager.cpp 132 init Peer connection unavailable, activating service
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "PendingRegistration" SignOn::IdentityImpl(0x10a63760)
../../../../lib/SignOn/identityimpl.cpp 172 storeCredentials Storing credentials
../../../../lib/SignOn/connection-manager.cpp 150 init Connected to "libsignon-qt2"
QUnifiedTimer::stopAnimationDriver: driver is not running
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "Ready" SignOn::IdentityImpl(0x10a63760)
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "NeedsUpdate" SignOn::IdentityImpl(0x10a63760)
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "PendingUpdate" SignOn::IdentityImpl(0x10a63760)
../../../../lib/SignOn/identityimpl.cpp 442 infoUpdated SERVER INFO UPDATED. NeedsUpdate " 0 "
../../../../lib/SignOn/identityimpl.cpp 345 storeCredentialsReply stored id: 12 old id: 0
../../../../lib/SignOn/identityimpl.cpp 242 queryInfo Querying info.
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "PendingUpdate" SignOn::IdentityImpl(0x10a63760)
../../../../lib/SignOn/identityimpl.cpp 381 getInfoReply QMap( ****** ) 
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "Ready" SignOn::IdentityImpl(0x10a63760)
Info:
    Id: 12
    caption: "ktp-morse-telegram"
    owner: ""
    userName: "******"
../../../Accounts/account-service.cpp: 162 onEnabled
../../../Accounts/account-service.cpp: 169 onChanged
11
11
Ignoring:  "IM"
../../../Accounts/account.cpp: 626 account_store_cb Saved accunt ID: 11
../../../../lib/SignOn/identityimpl.cpp 381 getInfoReply QMap( ****** ) 
../../../../lib/SignOn/identityimpl.cpp 96 updateState Updating state:  "Ready" SignOn::IdentityImpl(0x10a63760)
AccountsModel::accountCreated:  11
ktp-kded-module: 
QXcbConnection: XCB error: 3 (BadWindow), sequence: 37349, resource id: 144703522, major code: 15 (QueryTree), minor code: 0
../../../../lib/SignOn/connection-manager.cpp 171 onDisconnected Disconnected from daemon
../../../../lib/SignOn/async-dbus-proxy.cpp 218 setDisconnected

../../../../lib/SignOn/identityimpl.cpp 381 getInfoReply 
```

I was also able to generate a crash by clicking "configure account" in KDE Telepathy.

**Comment by Kaffeine** [comment]
What do you mean by KDE Telepathy? Can you name the package with the app? I would try install it and check, how it work.
I use ktp-contactlist application and have no SignOn or anything similar. There is just no place for red or any bubble in the contact-list application.
I have instant-messaging applet on the panel, but never seen any bubbles there to.

Does your telepathy installation work for any other connection manager?
For me, it doesn't looks like an issue in Morse, especially with such debug output (ConnectionManager successfully registered and have no any further requests, such as to set some account status to online).

I would ask in #kde-telepathy tomorrow, may be it's some known problem with SignOn or accounts setup.

I have no ktp-accounts-kcm ATM, but I just created a new account via `mc-tool add morse/telegram string:account=myphone; mc-tool enable morse/telegram/_3<myphone>0` and came online via the IM applet. Equally, ktp-contactlist could be run and global status could be set to Available.

**Comment by miabrahams** [comment]
OK, I also had a feeling that this may be an issue with the ecosystem, not just telepathy-morse.

By KDE Telepathy mean the KDE instant messenger that comes installed on Kubuntu.  That includes ktp-contactlist as well as a daemon providing a system tray feature and perhaps some other things.  In ktp-contact list, but when I go to "Instant Messaging Settings" and click "configure" I see a pop-up message  (like `kdialog --passivepopup "There was a problem trying to connect..."`) with a red error icon.  

I'm able to use Google Chat and other protocols correctly. Using the mc-tool commands you described seemed to have the same effect as doing "create new account" in ktp-accounts-kcm, and I see the same error messages. I will ask in #kde-telepathy to see if they can diagnose the problem.

**Comment by mck182** [comment]
The configure button crash is fixed in 15.08.2, please update.

Also, "setupSocketConnection p2p error: QDBusError("org.freedesktop.DBus.Error.FileNotFound", "Failed to connect to socket /run/user/1000/signond/socket: No such file or directory") 1"

As for the connection error, perhaps the account is misconfigured (on creation even). It would require someone to check the morse plugin for ktp-accounts-kcm what it actually does.
